src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -72,6 +78,13 @@ public sealed class RestoreManagerPackage : AsyncPackage
                 _buildEvents = dte.Events.BuildEvents;
                 _buildEvents.OnBuildBegin += BuildEvents_OnBuildBegin;
 
+                var solutionBuildManager = (IVsSolutionBuildManager5)await GetServiceAsync(typeof(SVsSolutionBuildManager));
+                if (solutionBuildManager != null)
+                {
+                    uint updateSolutionEventsCookie4;
+                    solutionBuildManager.AdviseUpdateSolutionEvents4(this, out updateSolutionEventsCookie4);
```

Comment:
Rather than drop the HRESULT, pass it to something like `ErrorHandler.ThrowOnFailure`